cub/cub/thread/thread_operators.cuh

```diff
@@ -382,93 +383,74 @@ struct ReduceByKeyOp
 
 #ifndef _CCCL_DOXYGEN_INVOKED // Do not document
 
+/***********************************************************************************************************************
+ * SIMD operators
+ **********************************************************************************************************************/
+
 namespace internal
 {
 
+_CCCL_HOST_DEVICE uint32_t simd_operation_is_not_supported_before_sm90();
+
 template <typename T>
 struct SimdMin
 {
-  static_assert(cub::detail::always_false<T>(), "Unsupported specialization");
+  static_assert(cub::detail::always_false_v<T>, "Unsupported specialization");
 };
 
 template <>
-struct SimdMin<::cuda::std::int16_t>
+struct SimdMin<int16_t>
 {
-  using simd_type = ::cuda::std::uint32_t;
-
-  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE ::cuda::std::uint32_t
-  operator()(::cuda::std::uint32_t a, ::cuda::std::uint32_t b) const
+  _CCCL_NODISCARD _CCCL_DEVICE _CCCL_FORCEINLINE uint32_t operator()(uint32_t a, uint32_t b) const
   {
-    return __vmins2(a, b);
+    NV_IF_TARGET(NV_PROVIDES_SM_90,
+                 (return __vmins2(a, b);), //
+                 (return simd_operation_is_not_supported_before_sm90();));
```

Comment:
I do not like us dropping support for earlier architectures. Especially given that it is a compile time optimization and should not be made a linker error